pkgs/by-name/ty/ty/package.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  installShellFiles,
+
+  buildPackages,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ty";
+  version = "0.0.1-alpha.1";
+
+  src = fetchFromGitHub {
+    owner = "astral-sh";
+    repo = "ruff";
+    rev = "f8890b70c35b2d05a8fa21958077de2aab5ba2ad";
```

Might be worth adding a comment explaining how to get this:
```suggestion
    # From the ruff submodule in the ty repo
    rev = "f8890b70c35b2d05a8fa21958077de2aab5ba2ad";
```